Automattic/pocket-casts-ios
# Referrals: Entry point

| 📘 Part of: #2083  |  
|:---:|

Fixes # 
This PR adds the following:
 - Adds a feature flag to control the display of referrals feature
 - Implements the entry point on the Profile View Controller and show the pass badge

~Note: The color of the badge is being worked on by design to correctly work on all themes~

| 1 | 2 | 3 | 4 | 5 |
| - | - | - | - | - |
| ![Simulator Screenshot - iPhone 15 Pro - 2024-08-23 at 14 53 42](https://github.com/user-attachments/assets/fab650cb-d2aa-4d8a-99f3-e9c9732ffb09) | ![Simulator Screenshot - iPhone 15 Pro - 2024-08-23 at 14 53 27](https://github.com/user-attachments/assets/3e977383-ab95-4fd6-b275-de68e349a645) |  ![Simulator Screenshot - iPhone 15 Pro - 2024-08-23 at 14 53 18](https://github.com/user-attachments/assets/b2b0bf91-afcd-4ca3-8915-2b296e9ec2a3) |  ![Simulator Screenshot - iPhone 15 Pro - 2024-08-23 at 14 53 09](https://github.com/user-attachments/assets/ca7b5f0b-c7ec-4a28-88be-759e831e178e) | ![Simulator Screenshot - iPhone 15 Pro - 2024-08-23 at 14 52 59](https://github.com/user-attachments/assets/1097bb44-30c5-43be-afd1-e928f290b5e0) |



## To test

- Start the app
- Ensure that have the Referrals FF active
- Go to Profile Tab
- If you are not logged in or don't have a subscription (Plus or Patron) you shouldn't see the gift icon on the top right
- Only if you are logged in and have a active Subscription you should see the gift icon
- Tap on the gift icon
- Check that the badge number overlayed on the gift icon  goes down
- Check that when the number goes down to zero the badge icon is hidden

## Checklist
